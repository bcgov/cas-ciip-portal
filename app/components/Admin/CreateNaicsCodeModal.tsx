import React from 'react';
import {Button, Form, Modal, Container, Col, Row, Alert} from 'react-bootstrap';

interface Props {
  validated: boolean;
  onSubmit: (form: any) => Promise<void>;
  show: boolean;
  onClose: () => void;
  showActiveCodeError: boolean;
  disabled?: boolean;
}

export const CreateNaicsCodeModal: React.FunctionComponent<Props> = ({
  validated,
  onSubmit,
  show,
  onClose,
  showActiveCodeError,
  disabled
}) => {
  const submitForm: (e: React.SyntheticEvent<any>) => void = async (e) => {
    console.log(e);
    e.stopPropagation();
    e.preventDefault();
    e.persist();

    await onSubmit(e.target);
    onClose();
  };

  return (
    <>
      <Modal
        centered
        size="lg"
        show={show}
        onHide={() => onClose()}
        aria-modal="true"
      >
        <Modal.Header closeButton>
          <Modal.Title>New NAICS Code</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Container>
            <Form noValidate validated={validated} onSubmit={submitForm}>
              <fieldset disabled={disabled}>
                <Row>
                  <Col md={4}>
                    <Form.Group controlId="naicsCode">
                      <Form.Label>NAICS Code</Form.Label>
                      <Form.Control required type="text" />
                      <Form.Control.Feedback type="invalid">
                        NAICS Code is required
                      </Form.Control.Feedback>
                    </Form.Group>
                  </Col>
                  <Col md={8}>
                    <Form.Group controlId="ciipSector">
                      <Form.Label>CIIP Sector</Form.Label>
                      <Form.Control type="text" />
                    </Form.Group>
                  </Col>
                </Row>
                <Form.Group controlId="naicsDescription">
                  <Form.Label>NAICS Description</Form.Label>
                  <Form.Control required type="text" />
                  <Form.Control.Feedback type="invalid">
                    NAICS Description is required
                  </Form.Control.Feedback>
                </Form.Group>
                {showActiveCodeError && (
                  <Alert variant="danger">
                    This NAICS code already exists. Please delete the currently
                    active code before entering new information
                  </Alert>
                )}
                <Button variant="primary" type="submit">
                  Submit
                </Button>
              </fieldset>
            </Form>
          </Container>
        </Modal.Body>
      </Modal>
      <style jsx global>{`
        .modal-header {
          color: white;
          background: #003366;
        }
        .modal-body {
          background: #f5f5f5;
        }
        .close {
          color: white;
        }
      `}</style>
    </>
  );
};

export default CreateNaicsCodeModal;
