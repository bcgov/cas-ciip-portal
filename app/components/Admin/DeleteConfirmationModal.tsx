import React from 'react';
import {Button, Modal} from 'react-bootstrap';

interface Props {
  deleteObject: {
    deleteName: string;
    deleteItem: string;
    deleteItemDescription: string;
  };
  handleDelete: (e: React.SyntheticEvent<any>) => Promise<void>;
  show: boolean;
  onClose: () => void;
}

export const DeleteConfirmationModal: React.FunctionComponent<Props> = ({
  deleteObject,
  handleDelete,
  show,
  onClose
}) => {
  return (
    <Modal
      centered
      size="sm"
      show={show}
      onHide={() => onClose()}
      aria-modal="true"
    >
      <Modal.Header closeButton>
        <Modal.Title style={{margin: 'auto'}}>
          Delete {deleteObject.deleteName}
        </Modal.Title>
      </Modal.Header>
      <Modal.Body style={{textAlign: 'center'}}>
        <p>Confirm delete {deleteObject.deleteName}:</p>
        <h5>{deleteObject.deleteItem}</h5>
        <p>{deleteObject.deleteItemDescription}</p>
        <Button onClick={handleDelete} variant="danger">
          Confirm Delete
        </Button>
      </Modal.Body>
    </Modal>
  );
};

export default DeleteConfirmationModal;
