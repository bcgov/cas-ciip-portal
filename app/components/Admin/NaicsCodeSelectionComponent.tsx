import Link from 'next/link';
import {useRouter} from 'next/router';
import React from 'react';
import {ListGroup} from 'react-bootstrap';

interface Props {
  naicsCodes: {code: string; description: string; id: string}[];
}

export const NaicsCodeSelectionComponent: React.FunctionComponent<Props> = ({
  naicsCodes
}) => {
  const router = useRouter();

  return (
    <>
      <div className="scrollable" tabIndex={0}>
        <ListGroup variant="flush">
          {naicsCodes.map(({code, description, id}) => (
            <Link
              key={id}
              prefetch={false}
              href={{
                pathname: router.pathname,
                query: {
                  ...router.query,
                  naicsCodeId: id
                }
              }}
            >
              <ListGroup.Item action active={router.query?.naicsCodeId === id}>
                {description}
                <br />
                <b>{code}</b>
              </ListGroup.Item>
            </Link>
          ))}
        </ListGroup>
      </div>
      <style jsx>
        {`
          .scrollable {
            overflow-y: scroll;
            max-height: calc(100vh - 108px - 60px - 2.4rem - 146px);
          }
          .scrollable::-webkit-scrollbar {
            -webkit-appearance: none;
            width: 8px;
          }
          .scrollable::-webkit-scrollbar-thumb {
            border-radius: 4px;
            background-color: rgba(0, 0, 0, 0.5);
            box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
          }
        `}
      </style>
    </>
  );
};
