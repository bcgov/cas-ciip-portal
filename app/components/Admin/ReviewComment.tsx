import React from 'react';
import {Button} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faTrash} from '@fortawesome/free-solid-svg-icons';
import {dateTimeFormat} from 'functions/formatDates';

interface Props {
  description: string;
  createdAt: string;
  createdBy: string;
  viewOnly: boolean;
}

export const ReviewComment: React.FunctionComponent<Props> = ({
  description,
  createdAt,
  createdBy,
  viewOnly
}) => {
  return (
    <li>
      <div className="text-block">
        <p>{description}</p>
        <p className="timestamp">
          <span>{createdBy}</span>
          {`${dateTimeFormat(createdAt, 'minutes')}`}
        </p>
      </div>
      {!viewOnly && (
        <div>
          <Button
            size="sm"
            variant="outline-primary"
            style={{marginRight: '0.5em'}}
          >
            Resolve
          </Button>
          <Button
            size="sm"
            aria-label="Delete"
            variant="secondary"
            title="Delete"
          >
            <FontAwesomeIcon icon={faTrash} />
          </Button>
        </div>
      )}
      <style jsx>{`
        li {
          list-style-type: none;
          padding: 0;
          margin-bottom: 1rem;
          width: 100%;
          display: inline-flex;
          justify-content: space-between;
        }
        li:last-of-type {
          margin: 0;
        }
        .text-block {
          max-width: calc(100% - 120px - 0.5rem);
        }
        .text-block ~ div {
          margin-left: 0.5rem;
        }
        p {
          margin-bottom: 0.5em;
          line-height: 1.2;
          font-size: 0.9rem;
        }
        .timestamp {
          font-size: 0.8rem;
          margin-bottom: 0;
          line-height: 1.5;
          color: #666;
        }
        .timestamp span {
          margin-right: 0.5em;
        }
      `}</style>
    </li>
  );
};

export default ReviewComment;
