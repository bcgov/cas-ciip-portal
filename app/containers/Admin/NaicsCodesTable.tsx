import React from 'react';
import {Table, Button} from 'react-bootstrap';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {NaicsCodesTable_query} from '__generated__/NaicsCodesTable_query.graphql';

interface Props {
  relay: RelayProp;
  query: NaicsCodesTable_query;
}

export const NaicsCodesTableComponent: React.FunctionComponent<Props> = (
  props
) => {
  const {query} = props;
  return (
    <>
      <div style={{textAlign: 'right'}}>
        <Button
          style={{marginTop: '-100px'}}
          // onClick={() => setDialogMode('create')}
        >
          New Naics Code
        </Button>
      </div>
      <Table striped bordered hover>
        <thead>
          <tr>
            <th scope="col">NAICS Code</th>
            <th scope="col">CIIP Sector</th>
            <th scope="col">NAICS Description</th>
            <th />
          </tr>
        </thead>
        <tbody>
          {query.allNaicsCodes.edges.map(({node}) => {
            return (
              <tr key={node.id}>
                <td>{node.naicsCode}</td>
                <td>{node.ciipSector}</td>
                <td>{node.naicsDescription}</td>
                <td>
                  <Button onClick={() => console.log('edit stuff')}>
                    Edit
                  </Button>
                </td>
              </tr>
            );
          })}
        </tbody>
      </Table>
    </>
  );
};

export default createFragmentContainer(NaicsCodesTableComponent, {
  query: graphql`
    fragment NaicsCodesTable_query on Query {
      allNaicsCodes(
        filter: {deletedAt: {equalTo: null}}
        orderBy: NAICS_CODE_DESC
      ) {
        edges {
          node {
            id
            naicsCode
            ciipSector
            naicsDescription
          }
        }
      }
    }
  `
});
