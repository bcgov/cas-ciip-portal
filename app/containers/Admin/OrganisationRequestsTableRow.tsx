import React from 'react';
import {Button} from 'react-bootstrap';
import {graphql, createFragmentContainer} from 'react-relay';
import {OrganisationRequestsTableRow_userOrganisation} from '__generated__/OrganisationRequestsTableRow_userOrganisation.graphql';
interface Props {
  userOrganisation: OrganisationRequestsTableRow_userOrganisation;
  key: string;
}
export const OrganisationRequestsTableRowComponent: React.FunctionComponent<
  Props
> = props => {
  console.log(props);
  const {userOrganisation} = props;
  return (
    <tr>
      <td>{userOrganisation.userId}</td>
      <td>{userOrganisation.userByUserId.firstName}</td>
      <td>{userOrganisation.userByUserId.lastName}</td>
      <td>{userOrganisation.organisationByOrganisationId.operatorName}</td>
      <td>{userOrganisation.status}</td>
      <td>
        <Button variant="success">Approve</Button>
      </td>
      <td>
        <Button variant="danger">Reject</Button>
      </td>
    </tr>
  );
};

export default createFragmentContainer(OrganisationRequestsTableRowComponent, {
  userOrganisation: graphql`
    fragment OrganisationRequestsTableRow_userOrganisation on UserOrganisation {
      status
      userId
      userByUserId {
        firstName
        lastName
      }
      organisationByOrganisationId {
        operatorName
      }
    }
  `
});
