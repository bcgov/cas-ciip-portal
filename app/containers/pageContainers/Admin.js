import React, {Component} from 'react';
import {graphql} from 'react-relay';
import {Row, Col, Jumbotron} from 'react-bootstrap';
import DefaultLayout from '../../layouts/default-layout';
import ProductCreatorContainer from '../Products/ProductCreatorContainer';
import ProductListContainer from '../Products/ProductListContainer';

class Admin extends Component {
  state = {
    formData: {formId: '', formJson: ''},
    mode: 'view',
    confirmationModalOpen: false
  };

  /** **  ProductRowItem Actions ** **/

  // Toggle enabling of editing products
  toggleProductMode = () => {
    this.state.mode === 'view' || this.state.mode === 'benchmark'
      ? this.setState({mode: 'product'})
      : this.setState({mode: 'view'});
  };

  // Toggle enabling of editing benchmarks
  toggleBenchmarkMode = () => {
    this.state.mode === 'view' || this.state.mode === 'product'
      ? this.setState({mode: 'benchmark'})
      : this.setState({mode: 'view'});
  };

  openConfirmationWindow = () => {
    this.setState({confirmationModalOpen: true});
  };

  closeConfirmationWindow = () => {
    this.setState({confirmationModalOpen: false});
  };

  formIdHandler = (formId, formJson) => {
    this.setState({formData: {formId, formJson}});
    console.log('form-builder.js > formIdHandler state', this.state);
  };

  productRowActions = {
    toggleProductMode: this.toggleProductMode,
    toggleBenchmarkMode: this.toggleBenchmarkMode,
    openConfirmationWindow: this.openConfirmationWindow,
    closeConfirmationWindow: this.closeConfirmationWindow
  };
  /** ** END ProductRowItem Actions ** **/

  static query = graphql`
    query AdminQuery {
      query {
        ...ProductListContainer_query
      }
    }
  `;

  render() {
    const {query} = this.props;
    return (
      <>
        <DefaultLayout title="Products and Benchmarks">
          <Row>
            <Col>
              <br />
              <Jumbotron>
                <h4>Create a Product</h4>
                <br />
                <ProductCreatorContainer />
              </Jumbotron>
              <br />
              <br />
              <br />
              <ProductListContainer
                productRowActions={this.productRowActions}
                query={query}
                mode={this.state.mode}
                confirmationModalOpen={this.state.confirmationModalOpen}
              />
            </Col>
          </Row>
        </DefaultLayout>
      </>
    );
  }
}

export default Admin;

/*

Product Creator:
1: Add table for product (name and description)
2: Add component for createproduct
3: Add add fields for create product in component: Product name and description
4: On save create object and push to createProduct mutation

List of products
1: create component for List of Products
2: use queryrenderer to loop through products
3: display products as a list

BM and ET
1: Create a component for BM and ET
2: Pass product object to the component
3: Component has prdouct name, description, BM and ET fields and a save button
4: on save create update BM/ET Values: Todo: Add history in the future
5: Update the components

 */
