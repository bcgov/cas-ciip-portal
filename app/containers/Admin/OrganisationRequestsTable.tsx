import React, {useEffect} from 'react';
import {
  Table,
  Container,
  Dropdown,
  Button,
  Row,
  Col,
  Form
} from 'react-bootstrap';
import {graphql, createRefetchContainer, RelayRefetchProp} from 'react-relay';
import {OrganisationRequestsTable_query} from '__generated__/OrganisationRequestsTable_query.graphql';
import DropdownMenuItemComponent from '../../components/DropdownMenuItemComponent';
import OrganisationRequestsTableRow from './OrganisationRequestsTableRow';
interface Props {
  query: OrganisationRequestsTable_query;
  orderByField: string;
  orderByDisplay: string;
  searchField: string;
  searchValue: string;
  direction: string;
  searchDisplay: string;
  handleEvent: (...args: any[]) => void;
  relay: RelayRefetchProp;
}
export const OrganisationRequestsTableComponent: React.FunctionComponent<
  Props
> = props => {
  const {
    orderByField,
    orderByDisplay,
    searchField,
    searchValue,
    searchDisplay,
    direction,
    handleEvent
  } = props;
  const dropdownSortItems = [
    {eventKey: 'user_id', title: 'User ID'},
    {eventKey: 'first_name', title: 'First Name'},
    {eventKey: 'last_name', title: 'Last Name'},
    {eventKey: 'status', title: 'Status'},
    {eventKey: 'operator_name', title: 'Operator'}
  ];
  const {edges} = props.query.searchUserOrganisation;
  useEffect(() => {
    const refetchVariables = {
      searchField,
      searchValue,
      orderByField,
      direction
    };
    props.relay.refetch(refetchVariables);
  });

  return (
    <>
      <Container style={{padding: 20, background: '#dee2e6'}}>
        <Row>
          <Col md={3}>
            <h5>Sort User Requests</h5>
          </Col>
          <Col md={1} />
          <Col md={3}>
            <h5>Filter User Requests</h5>
          </Col>
        </Row>
        <Row>
          <Col md={2}>
            <Dropdown style={{width: '100%'}}>
              <Dropdown.Toggle
                style={{width: '100%', backgroundColor: '#036'}}
                id="dropdown-sort"
              >
                {orderByDisplay}
              </Dropdown.Toggle>
              <Dropdown.Menu style={{width: '100%'}}>
                {dropdownSortItems.map(item => (
                  <DropdownMenuItemComponent
                    key={item.eventKey}
                    itemId="sortApplications"
                    itemEventKey={item.eventKey}
                    itemFunc={(eventKey, event) => handleEvent(event, eventKey)}
                    itemTitle={item.title}
                  />
                ))}
                ;
              </Dropdown.Menu>
            </Dropdown>
          </Col>
          <Col md={1}>
            <Button
              id="toggleDirection"
              style={{width: '100%', backgroundColor: '#036'}}
              onClick={event => handleEvent(event)}
            >
              {direction}
            </Button>
          </Col>
          <Col md={1} />
          <Col md={2}>
            <Dropdown style={{width: '100%'}}>
              <Dropdown.Toggle
                style={{width: '100%', backgroundColor: '#036'}}
                id="dropdown-filter"
              >
                {searchDisplay}
              </Dropdown.Toggle>
              <Dropdown.Menu style={{width: '100%'}}>
                {[
                  {eventKey: null, title: 'No Filter'},
                  ...dropdownSortItems
                ].map(item => (
                  <DropdownMenuItemComponent
                    key={item.eventKey}
                    itemId="applySearchField"
                    itemEventKey={item.eventKey}
                    itemTitle={item.title}
                    itemFunc={(eventKey, event) => handleEvent(event, eventKey)}
                  />
                ))}
                ;
              </Dropdown.Menu>
            </Dropdown>
          </Col>
          <Col md={6}>
            <Form id="applySearchValue" onSubmit={event => handleEvent(event)}>
              <Form.Row>
                <Form.Group as={Col} md={8}>
                  <Form.Control type="string" />
                </Form.Group>
                <Form.Group as={Col} md={1}>
                  <Button type="submit" style={{backgroundColor: '#036'}}>
                    Filter
                  </Button>
                </Form.Group>
              </Form.Row>
            </Form>
          </Col>
        </Row>
      </Container>
      <br />
      <br />
      <Table
        striped
        hover
        style={{textAlign: 'center', border: '1px solid #f5f5f5'}}
      >
        <thead style={{backgroundColor: '#036', color: 'white'}}>
          <tr>
            <th>#</th>
            <th>Name</th>
            <th>Lastname</th>
            <th>Operator Requested</th>
            <th>Status</th>
            <th />
            <th />
          </tr>
        </thead>
        <tbody>
          {edges.map(edge => (
            <OrganisationRequestsTableRow
              key={edge.node.id}
              userOrganisation={edge.node}
            />
          ))}
        </tbody>
      </Table>
    </>
  );
};

export default createRefetchContainer(
  OrganisationRequestsTableComponent,
  {
    query: graphql`
      fragment OrganisationRequestsTable_query on Query
        @argumentDefinitions(
          searchField: {type: "String"}
          searchValue: {type: "String"}
          orderByField: {type: "String"}
          direction: {type: "String"}
        ) {
        searchUserOrganisation(
          searchField: $searchField
          searchValue: $searchValue
          orderByField: $orderByField
          direction: $direction
        ) {
          edges {
            node {
              id
              ...OrganisationRequestsTableRow_userOrganisation
            }
          }
        }
      }
    `
  },
  graphql`
    query OrganisationRequestsTableRefetchQuery(
      $searchField: String
      $searchValue: String
      $orderByField: String
      $direction: String
    ) {
      query {
        ...OrganisationRequestsTable_query
          @arguments(
            searchField: $searchField
            searchValue: $searchValue
            orderByField: $orderByField
            direction: $direction
          )
      }
    }
  `
);
