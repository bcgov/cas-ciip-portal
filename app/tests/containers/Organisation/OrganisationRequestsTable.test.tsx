import React from 'react';
import {shallow} from 'enzyme';
import {OrganisationRequestsTableComponent} from '../../../containers/Admin/OrganisationRequestsTable';

describe('Organisations', () => {
  it("should render the user's requested organisations", async () => {
    const query = {
      searchUserOrganisation: {
        edges: [
          {
            node: {
              id: 1,
              status: 'APPROVED'
            }
          }
        ]
      }
    };
    // @ts-ignore
    const r = shallow(<OrganisationRequestsTableComponent query={query} />);
    expect(r).toMatchSnapshot();
    expect(
      r
        .find('Relay(OrganisationRequestsTableRowComponent)')
        .prop('userOrganisation')
    ).toBe(query.searchUserOrganisation.edges[0].node);
  });
});
