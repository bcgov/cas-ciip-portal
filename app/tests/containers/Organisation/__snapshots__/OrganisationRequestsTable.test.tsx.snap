// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisations should render the user's requested organisations 1`] = `
<Fragment>
  <SearchTableLayoutComponent
    dropdownSortItems={
      Array [
        Object {
          "eventKey": "user_id",
          "title": "User ID",
        },
        Object {
          "eventKey": "first_name",
          "title": "First Name",
        },
        Object {
          "eventKey": "last_name",
          "title": "Last Name",
        },
        Object {
          "eventKey": "email_address",
          "title": "Email",
        },
        Object {
          "eventKey": "status",
          "title": "Status",
        },
        Object {
          "eventKey": "operator_name",
          "title": "Operator",
        },
      ]
    }
  />
  <br />
  <br />
  <Bootstrap(Table)
    hover={true}
    striped={true}
    style={
      Object {
        "border": "1px solid #f5f5f5",
        "textAlign": "center",
      }
    }
  >
    <thead
      style={
        Object {
          "backgroundColor": "#036",
          "color": "white",
        }
      }
    >
      <tr>
        <th>
          #
        </th>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
        <th>
          Email
        </th>
        <th>
          Operator Requested
        </th>
        <th>
          Status
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <Relay(OrganisationRequestsTableRowComponent)
        key="1"
        userOrganisation={
          Object {
            "id": 1,
            "status": "APPROVED",
          }
        }
      />
    </tbody>
  </Bootstrap(Table)>
</Fragment>
`;
